src/saving/ISaveUpgradeStep.cs

```diff
@@ -162,6 +163,174 @@ private void UpgradeBehaviouralValues(JProperty property, JEnumerable<JProperty>
+            if (property.Name == "Adjacent")
+            {
+                foreach (var adjacent in property.Value)
+                {
+                    var adjacentChildren = adjacent.Children<JProperty>();
+                    var adjacentChildrenNames = adjacentChildren.Select(p => p.Name).ToList();
+
+                    if (adjacentChildrenNames.Contains("Depth") && !adjacentChildrenNames.Contains("Volume"))
+                    {
+                        UpgradePatchesVolume((JObject)adjacent, adjacentChildren);
+                    }
+                }
+            }
+
+            // Add gas production effect in effects array
+            if (property.Name == "effects")
+            {
+                var gasProductionEffectName = "GasProductionEffect, Thrive";
+                if (!PropertyHasNamedEffect(property, gasProductionEffectName))
+                {
+                    UpgradeGasProductionEffect(property, gasProductionEffectName);
+                }
+            }
+        }
+
+        private bool PropertyHasNamedEffect(JProperty property, string name)
+        {
+            if (property.Value is JArray tokenArray)
+            {
+                foreach (var effect in tokenArray)
+                {
+                    var effectProperties = effect.Children<JProperty>();
+
+                    if (effectProperties.Select(p => p.Value.ToString()).Contains(name))
+                        return true;
+                }
+            }
+
+            return false;
+        }
+
+        /// <summary>
+        ///   Updates patches values by adding volume to it as a cube.
+        /// </summary>
+        /// <param name="container">The JObject containing the patch definition.</param>
+        /// <param name="children">The children of this JObject.</param>
+        /// <remarks>
+        ///   <para>
+        ///     Changes a json either like:
+        ///       "0": {
+        ///         ...
+        ///         "Depth": [2500, 3000],
+        ///         ...
+        ///       }
+        ///     or
+        ///       "Adjacent": [
+        ///         {
+        ///           ...
+        ///           "Depth": [2500, 3000],
+        ///           ...
+        ///         },
+        ///         ...
+        ///       ]
+        ///     to
+        ///       ...
+        ///       {
+        ///         ...
+        ///         "Depth": [2500, 3000],
+        ///         "Volume": (3000 - 2500)^3,
+        ///         ...
+        ///       }
+        ///       ...
+        ///   </para>
+        /// </remarks>
+        private void UpgradePatchesVolume(JObject container, JEnumerable<JProperty> children)
+        {
+            if (children.First(p => p.Name == "Depth").Value is JArray depth)
+            {
+                if (depth[1].Value<int?>() == null || depth[0].Value<int?>() == null)
+                {
+                    throw new ArgumentException("Depth array does not start with two integers!");
```

Comment:
Should throw json exceptions like the other upgrade steps as these are about invalid json data.